### DISCUSSION TITLE
Vtk View using RouterViewLayout

### DISCUSSION THREAD

**Discussion by oscarBarEs** [Kitware/trame#462] [Q&A] [answered]
Hi, I am trying to create a trame app where i have 2 subPages, one where I show and vtk file and other where i show some related charts.

I dont have a problem showing the vtk on the main page, but when i put the same lines of code on a RouterViewLayout the view suddenly dissapears.

Here's and example:

    def _update_UI(self):
        with RouterViewLayout(self.server, "/"):

        with RouterViewLayout(self.server, "/heart"):
            with vtkTrame.VtkLocalView(self.pl.ren_win)as local:
                    def view_update(**kwargs):
                        local.update(**kwargs)                
            self.ctrl.view_update = view_update

        with RouterViewLayout(self.server, "/data"):     
            with vuetify.VContainer(classes="justify-left ma-6"):
                vuetify.VDataTable(**table_of_simulation(self.data))

        with SinglePageWithDrawerLayout(self.server) as layout:
            with layout.toolbar:
                self.header()
            with layout.content:
                # Here vtkTrame.VtkLocalView(self.pl.ren_win) shows
                router.RouterView()
         return layout

**Comment by jourdain** [comment, marked as answer]
That is because your router content has no size. You may need to do something like

```python
with RouterViewLayout(self.server, "/heart") as layout:
        layout.root.classes="fill-height"
        with vtkTrame.VtkLocalView(self.pl.ren_win)as local:
                def view_update(**kwargs):
                    local.update(**kwargs)                
        self.ctrl.view_update = view_update
```

You can also see my [comment on that thread](https://github.com/Kitware/trame/discussions/444).

**Reply by oscarBarEs** [reply]
Thank you so so much!! That was it :))
